plugin/src/App/getTextBounds.lua

```diff
@@ -0,0 +1,26 @@
+local TextService = game:GetService("TextService")
+
+local Rojo = script:FindFirstAncestor("Rojo")
+local Packages = Rojo.Packages
+
+local Log = require(Packages.Log)
+
+local params = Instance.new("GetTextBoundsParams")
+
+local function getTextBounds(text: string, font: Font, textSize: number, width: number, richText: boolean?): Vector2
```

Comment:
This function can yield, so it should probably have a name that reflects this - and it's also not OK to yield in Roact! I'm not sure exactly *when* it will yield, and our `TextService:GetTextBoundsAsync` usage has seemed fine so far, but it probably warrants further review. You're more familiar with this area of the codebase so you'll be more able to tell if it's a problem or not

Reply:
`GetTextBoundsAsync` only yields when the font isn't loaded, as per [Roblox's own docs](https://create.roblox.com/docs/reference/engine/classes/TextService#GetTextBoundsAsync). Assuming we trust them, we could ensure it never yields by preloading the fonts when the plugin initializes.

That said, I suspect that builtin fonts are just always loaded in Studio or load quick enough that it won't matter. This would be more of a concern if we were using a custom font.